server_spec.rb

```diff
@@ -0,0 +1,234 @@
+require_relative 'server'
```

Comment:
Prefer to move test files to a `spec` folder so Rspec can find them automatically, and move code to `lib` for the same reason (and so we can remove `require_relative`.)